### ISSUE TITLE
Issue with React deployments - Segments does not support...

### ISSUE THREAD

**Issue by mmontesinoademero** [Hexastack/eazychart#107]
Everything works great in development, but when deploying my React app, the app is throwing the error "Segments does not support non ordinal color scale" due to the deployed eazychart.esm.js, specifically the call to...

if("ScaleOrdinal"===a.constructor.name)

...which corresponds to [line 35 at ez-react/src/components/Segments.tsx.](https://github.com/Hexastack/eazychart/blob/2b006cdee429e981cf8a5ce1a3ae2e1de34a1220/packages/ez-react/src/components/Segments.tsx#L35)

If I change this to...

a instanceof ScaleOrdinal (eazychart.esm.js)
or
colorScale instanceof ScaleOrdinal (Segments.tsx)

...then everything works as expected. I've not yet dug into see if this is related to my own Babel/Webpack configuration. I've been using the same deployment scripts for many years and never seen anything like this.

**Comment by marrouchi** [comment]
Hello @mmontesinoademero, thanks for reporting this, we will address the issue as soon as possible :)

**Comment by marrouchi** [comment]
If I remember correctly we had a typescript issue when using : 
```
a instanceof ScaleOrdinal (eazychart.esm.js)
or
colorScale instanceof ScaleOrdinal (Segments.tsx)
```

**Comment by marrouchi** [comment]
@mmontesinoademero  The root cause for this issue is well detailed here  : https://www.dannyguo.com/blog/how-to-fix-instanceof-not-working-for-custom-errors-in-typescript

The `OrdinalScale` class extends the `AstractScale` class and we perform an "instanceof", in storybook : 
```
const scale = new ScaleOrdinal( ... );
console.log(scale instanceof ScaleOrdinal); // prints false
console.log(scale instanceof AbstractScale); // prints false 

This  seems to be an issue if when the compilation target is ES3 or ES5, but the wierd thing is that the configured target we have is "ESNEXT". Apparently "Storybook" relies on Babel and does not account apparently for that config.

TL;DR
We did remove the type safeguard for the moment so that we would think of a better way to handle this in the future : https://github.com/Hexastack/eazychart/pull/108 a new release is on the way ;)

**Comment by marrouchi** [comment]
@mmontesinoademero A new version `0.3.1-alpha.0` has been published, please let us know if you encounter other issues :)
